LinkedList/LinkedList/TSLList.h

```diff
@@ -0,0 +1,352 @@
+
+		for (int i = 0; i < Size; i++)
+		{
+			AddTail(Temporal->Element);
+			Temporal = Temporal->Next;
+		}
+		return *this;
+
+	}
+
+	//-----------------------------------------------
+	Type& GetHead()
+	{
+		return Head->Element;
+	}
+
+	const Type& GetHead() const
+	{
+		return Head->Element;
+	}
+
+	//-----------------------------------------------
+	Type& GetTail()
+	{
+		return Tail->Element;
+	}
+
+	const Type& GetTail() const
+	{
+		return Tail->Element;
+	}
+
+	//---------------------------------Size
+	bool IsEmpty()
+	{
+		return Size == 0;
+	}
+
+	//------------------------------------------------
+	int GetSize()
+	{
+		return Size;
+	}
+
+	//-----------------------------------Modifiers
+	void AddHead(const Type& InHead)
+	{
+		FNode* NewHead = new FNode;
+		NewHead->Element = InHead;
+
+		if (Head == nullptr)
+		{
+			Head = NewHead;
+			Tail = NewHead;
+			Size++;
+		}
+		else
+		{
+			NewHead->Next = Head;
+			Head = NewHead;
+			Size++;
+		}
+	}
+
+	//-------------------------------------------------
+	void AddTail(const Type& InTail)
+	{
+
+		if (Head == nullptr)
+		{
+			AddHead(InTail);
+		}
+
+		else
+		{
+			Tail->Next = new FNode;
+			Tail->Next->Element = InTail;
+			Tail = Tail->Next;
+			Size++;
+		}
+	}
+	
+	//----------------------------------------------------
+	void Insert(const Type& NewItem, const int Index)
+	{
+		if (Index == 0)
+		{
+			AddHead(NewItem);
+		}
+		else if(Index == Size)
+		{
+			AddTail(NewItem);
+		}
+
+		else if (Index > 0 && Index < Size)
+		{
+			FNode* Temporal = Head;
+			FNode* Container = new FNode;
+
+			for (int i = 0; i < Index; i++)
+			{
+				Temporal = Temporal->Next;
+			}
+			Container->Element = Temporal->Element;
+
+			Temporal->Element = NewItem;
+			Container->Next = Temporal->Next;
+			Temporal->Next = Container;
+			Size++;
+		}
+	}
+	//---------------------------------------------------------
+	void RemoveHead()
+	{
+		if (Size != 0)
+		{
+			FNode* NewHead = Head->Next;
+			delete Head;
+			Head = NewHead;
+			--Size;
+		}
+
+		else if (Size == 1)
+		{
+			delete Head;
+			Head = nullptr;
+			Tail = nullptr;
+			--Size;
+		}
+	}
+
+	//---------------------------------------------------------
+	void Remove(const int Index)
+	{
+		FNode* Temporal = Head;
+
+		if (Index >= 0 && Index <= Size)
+		{
+			for (int i = 0; i < Index - 1; i++)
+			{
+				Temporal = Temporal->Next;
+			}
+
+			FNode* Current = Temporal->Next;
+			Temporal->Next = Current->Next;
+
+			if (Current == Tail)
+			{
+				Tail = Temporal;
+			}
+			delete Current;
+			
+			Size--;
+		}
+	}
+
+	//----------------------------------------------------------
+	void Clear()
+	{
+		FNode* NextItem;
+
+		for (int i = 0; i < Size; i++)
+		{
+			NextItem = Head->Next;
+			delete Head;
+			Head = NextItem;
+		}
+		Size = 0;
+		Head = nullptr;
+		Tail = nullptr;
+	}
+
+	//----------------------------------------ForEach
+	template<typename Function>
+	void forEach(const Function& SomeFunction)
```

Comment:
I also initially used SomeFunction here, as that's what JD used during the lecture. However, on Discord he posted the updated version of the signature, and ideally it should be called "Pred" for the typename and "Predicate" for the parameter name, like you have below.

Also, the F in ForEach should be capitalized, I think.